girder/models/model_base.py

```diff
@@ -90,7 +90,7 @@ def reconnect(self):
         typically not have to call this method.
         """
         db_connection = getDbConnection()
-        self.database = db_connection.get_default_database()
+        self.database = db_connection.get_database()
```

Comment:
I see you've increased the pymongo version to allow for this (since it won't work with pymongo < 3.4).  Is there any other benefit to a newer pymongo?

Reply:
First-class compatibility with `mongod` 3.6.